Sources/WordPressKit/Services/ActivityServiceRemote.swift

```diff
@@ -152,18 +152,15 @@ open class ActivityServiceRemote: ServiceRemoteWordPressComREST {
                                         failure(ResponseError.decodingFailure)
                                     }
                                 }, failure: { error, _ in
-                                    // FIXME: A hack to support free WPCom sites and Rewind. Should be obsolote as soon as the backend
-                                    // stops returning 412's for those sites.
-                                    let nsError = error as NSError
-
-                                    guard nsError.domain == WordPressComRestApiEndpointError.errorDomain,
-                                       nsError.code == WordPressComRestApiErrorCode.preconditionFailure.rawValue else {
-                                        failure(error)
+                                    // FIXME: A hack to support free WPCom sites and Rewind.
+                                    // Should be obsolete as soon as the backend stops returning 412's for those sites.
+                                    guard error.castedToEndpointErrorWitCode(.preconditionFailure) != nil else {
+                                        success(RewindStatus(state: .unavailable))
                                         return
                                     }
```

Comment:
My brain can't process these two guard statements. But, are their logic changed?

Reply:
I noticed an unit test failed, I think this guard change here might be the cause?